forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindingStatus.kt

```diff
@@ -6,22 +6,53 @@
  */
 package org.forgerock.android.auth.devicebind
 
+import java.security.PrivateKey
+
 /**
  * State of the Device Binding errors
  */
 
-sealed class DeviceBindingStatus(var message: String, val clientError: String, val errorCode: Int? = null)
-data class Timeout(private val errorMessage: String = "Biometric Timeout", private val errorType: String = "Timeout"): DeviceBindingStatus(errorMessage, errorType)
-data class Abort(private val errorMessage: String = "User Terminates the Authentication",  private val errorType: String = "Abort", private val code: Int? = null): DeviceBindingStatus(errorMessage, errorType, code)
-data class Unsupported(private var errorMessage: String? = "Device not supported. Please verify the biometric or Pin settings", private val errorType: String = "Unsupported"):
-    DeviceBindingStatus(errorMessage ?: "Failed to generate keypair or sign the transaction", errorType)
-data class UnRegister(private val errorMessage: String = "PublicKey or PrivateKey Not found in Device", private val errorType: String = "Unsupported"): DeviceBindingStatus(errorMessage, errorType)
-object Success: DeviceBindingStatus("", "")
+sealed class DeviceBindingStatus<out T : Any>(var message: String?,
```

Comment:
do we need this Generic ? trying to understand the purpose of T.

Reply:
We use DeviceBindingStatus for both, error and success, for error we would like to provide error message, error type and code, but for success, I am trying to return the authenticated PrivateKey, so that we can perform the signing after.

Reply:
i see , so this is just to differentiate the type...